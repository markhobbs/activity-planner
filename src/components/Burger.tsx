import React from "react";
import styled from 'styled-components';


const StyledBurger = styled.button<{ open: boolean }>`
  position: absolute;
  top: 1rem;
  left: 1rem;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  width: 2rem;
  height: 2rem;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0;
  z-index: 3;

  &:focus {
    outline: none;
  }

  div {
    width: 2rem;
    height: 0.25rem;

    background: ${({ open }) => open ? '#0D0C1D' : '#0D0C1D'};
    border-color: ${({ open }) => open ? '#0D0C1D' : '#0D0C1D'};

    border-radius: 10px;
    transition: all 0.3s linear;
    position: relative;
    transform-origin: 1px;

    :first-child {
      transform: ${({ open }) => open ? 'rotate(45deg)' : 'rotate(0)'};
    }

    :nth-child(2) {
      opacity: ${({ open }) => open ? '0' : '1'};
      transform: ${({ open }) => open ? 'translateX(20px)' : 'translateX(0)'};
    }

    :nth-child(3) {
      transform: ${({ open }) => open ? 'rotate(-45deg)' : 'rotate(0)'};
    }
  }
`

interface BurgerProps {
  open: boolean;
  setOpen: (open: boolean) => void;
}

const Burger: React.FC<BurgerProps> = ({ open, setOpen }) => {
  return (
    <StyledBurger 
      data-testid="burger-button"
      open={open} 
      onClick={() => setOpen(!open)}>
      <div />
      <div />
      <div />
    </StyledBurger>
  );
};

export default Burger;
