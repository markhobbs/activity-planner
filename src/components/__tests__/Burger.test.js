import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Burger from '../Burger';

describe('<Burger /> tests', () => {
    it('should match snapshot', () => {
        const { asFragment } = render(<Burger open={false} setOpen={() => {}} />);
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('<Burger /> tests', () => {
    it('should call setOpen when clicked', () => {
      const setOpenMock = jest.fn();
      const { getByTestId } = render(<Burger open={false} setOpen={setOpenMock} />);
  
      const burgerButton = getByTestId('burger-button');
      fireEvent.click(burgerButton);
  
      expect(setOpenMock).toHaveBeenCalledTimes(1);
    });
});