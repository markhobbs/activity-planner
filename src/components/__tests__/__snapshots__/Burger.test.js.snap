// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Burger /> tests should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-beySPh hiCbqQ"
    data-testid="burger-button"
  >
    <div />
    <div />
    <div />
  </button>
</DocumentFragment>
`;
